packages/@aws-cdk/aws-codepipeline/test/pipeline.test.ts

```diff
@@ -566,3 +859,57 @@ class ReusePipelineStack extends cdk.Stack {
     });
   }
 }
+
+const createPipeline = (scope: Construct, pipelineName?: string, pipelineId: string = 'Pipeline') => {
+  const artifact = new codepipeline.Artifact();
+  return new codepipeline.Pipeline(scope, pipelineId, {
+    pipelineName,
+    crossAccountKeys: true,
+    reuseCrossRegionSupportStacks: false,
+    stages: [
+      {
+        stageName: 'Source',
+        actions: [new FakeSourceAction({ actionName: 'Source', output: artifact })],
+      },
+      {
+        stageName: 'Build',
+        actions: [new FakeBuildAction({ actionName: 'Build', input: artifact })],
+      },
+    ],
+  });
+};
+
+interface CreatePipelineStackOptions {
+  context: cdk.App,
+  suffix: string,
+  stackId?: string,
+  pipelineId?: string,
+  undefinedStackName?: boolean,
+  nestedStackId?: string,
```

Comment:
Adding `readonly` to the properties here might be nice (although not required).

Reply:
Added